Assets/Scripts/StealthGame/Enemy.cs

```diff
@@ -0,0 +1,176 @@
+            waypoints[i] = new Vector3(waypoints[i].x, transform.position.y, waypoints[i].z);
+        }
+
+        StartCoroutine(PatrolPath(waypoints));
+    }
+
+    void Update()
+    {
+        if (CanSeePlayer())
+        {
+            spotLight.color = Color.red;
+            sgm.playerInput.DeactivateInput();
+            
+            if (OnPlayerSpotted != null)
+            {
+                OnPlayerSpotted();
+            }
+            Time.timeScale = 0;
+        }
+        else
+        {
+            spotLight.color = originalSpotlightColor;
+        }
+    }
+
+    #endregion
+
+    #region Methods
+
+    bool CanSeePlayer()
+    {
+        if (Vector3.Distance(transform.position, player.position) < viewDistance)
+        {
+            Vector3 directionToPlayer = (player.position - transform.position).normalized;
+            float angleToPlayer = Vector3.Angle(transform.forward, directionToPlayer);
+            if (angleToPlayer < viewAngle / 2f)
+            {
+                if (!Physics.Linecast(transform.position, player.position, viewMask))
+                {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    IEnumerator PatrolPath(Vector3[] waypoints)
+    {
+        transform.position = waypoints[0];
+        int targetWaypointIndex = 1;
+        Vector3 targetWaypoint = waypoints[targetWaypointIndex];
+        transform.LookAt(targetWaypoint);
+
+        while (true)
+        {
+            transform.position = Vector3.MoveTowards(transform.position, targetWaypoint, speed * Time.deltaTime);
+            if (transform.position == targetWaypoint)
+            {
+                targetWaypointIndex = (targetWaypointIndex +1) % waypoints.Length;
+                targetWaypoint = waypoints[targetWaypointIndex];
+                yield return new WaitForSeconds(waitTime);
+                yield return StartCoroutine(FaceTarget(targetWaypoint));
+            }
+            yield return null;
+        }
+    }
+
+    IEnumerator FaceTarget(Vector3 lookAtTarget)
+    {
+        Vector3 directionToLook = (lookAtTarget - transform.position).normalized;
+        float targetAngle = 90 - Mathf.Atan2(directionToLook.z, directionToLook.x) * Mathf.Rad2Deg;
+        while (Mathf.Abs(Mathf.DeltaAngle(transform.eulerAngles.y, targetAngle)) > 0.05f)
+        {
+            float angle = Mathf.MoveTowardsAngle(transform.eulerAngles.y, targetAngle, turnSpeed * Time.deltaTime);
+            transform.eulerAngles = Vector3.up * angle;
+            yield return null;
+        }
+    }
+
+    public override void Interact(Transform transform)
+    {
+        if (canInteract)
+        {
+            canInteract = false;
+            if (hasItem)
+            {
+                hasItem = false;
+                sgm.playerHasItem = true;
+                Debug.Log("Collected: " + item.name, gameObject);
+                Destroy(item);
+                StartCoroutine(EndInteract());
```

Comment:
We should make the enemy face the player when they grab the item as well. I think we should randomly call FaceTarget so the enemies look like they are looking around or they look left and right as they patrol to add more difficulty.